enhancements/sig-architecture/76-addon-install-strategy/README.md

```diff
@@ -0,0 +1,182 @@
+#### Story 3
+
+The addon developer should not care about which cluster the `ManagedClusterAddon` should be installed.
+
+## Design Details
+
+### ClusterManagementAddon API change
+
+```go
+type InstallStrategy struct {
+	// Type is the type of the install strategy, it can be:
+	// - Manual: no automatic install
+	// - ClusterLabelSelector: install to clusters with certain labels
+	// - Placements: install to clusters selected by placements.
+	// +kubebuilder:validation:Enum=Manual;Placements
+	// +kubebuilder:default:=Manual
+	Type string `json:"type"`
+
+	// ClusterLabelSelector is a label selector honored when install strategy type is
+	// ClusterLabelSelector
+	ClusterLabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// Placements is a list of placement references honored when install strategy type is
+	// Placements. All clusters selected by these placements will install the addon
+	Placements []PlacementRef `json:"placements,omitempty"`
+}
+
+type PlacementRef struct {
+	// Name of the placement
+	// +required
+	Name string `json:"name"`
+
+	// Namespace of the placement
+	// +required
+	Namespace string `json:"space"`
+
+	// AddonTemplate is the template to generate ManagedClusterAddon
+	AddonTemplate ManagedClusterAddOnSpec `json:"addonTemplate,omitempty"`
+}
+```
+
+### Install addon according to a placement
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: aws-placement
+      namespace: default
+```
+
+- when a cluster is added into the related `PlacementDecision`, the `ManagedClusterAddon` will be created on the cluster namespace.
+- when a cluster is removed from the related `PlacementDecision`, the `ManagedClusterAddon` will be deleted on the cluster namespace.
+
+### Install addons based on multiple placement with different configuration
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: aws-placement
+      namespace: default
+	  addonTemplate:
+	   configs:
+	   - group: addon.open-cluster-management.io
+	     resource: addondeploymentconfigs
+		 name: aws-config
+	- name: gcp-placement
+	  namespace: default
+	  addonTemplate:
+	   configs:
+	   - group: addon.open-cluster-management.io
+	     resource: addondeploymentconfigs
+		 name: gcp-config
+```
+
+This will ensure that addon installed in aws uses aws-config will addon installed in gcp uses gcp config.
+
+
+### Disable addon in some clusters
+
+At first change the install strategy to `Manual`
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Manual
+```
+
+Then admin can delete `ManagedClusterAddon` in certain cluster namespaces.
```

Comment:
if we remove some placements from clustermanagementaddon placements, the related addons will be deleted automatically, right?

Reply:
if we set the type to placement firstly, then we change it to manual, what will be happened? vice versa

Reply:
we do not automatically create/delete addon any longer, anything currently in the system is kept